test/submit_05_reorg_test.go

```diff
@@ -0,0 +1,147 @@
+
+	// prepare txStale
+	node_client.SendToAddress(t, bitcoind, address, float64(0.003))
+
+	utxos = node_client.GetUtxos(t, bitcoind, address)
+	require.True(t, len(utxos) > 0, "No UTXOs available for the address")
+
+	// make sure to pick the correct UTXO
+	for _, u := range utxos {
+		if u.Amount == float64(0.003) {
+			utxo = u
+		}
+	}
+
+	txStale, err := node_client.CreateTx(privateKey, address, utxo)
+	require.NoError(t, err)
+
+	// post a tx to the STALE chain
+	rawTx, err = txStale.EFHex()
+	require.NoError(t, err)
+	resp = postRequest[TransactionResponse](t, arcEndpointV1Tx, createPayload(t, TransactionRequest{RawTx: rawTx}), map[string]string{"X-WaitFor": StatusSeenOnNetwork}, http.StatusOK)
+	require.Equal(t, StatusSeenOnNetwork, resp.TxStatus)
+
+	// post the previously mined tx1 to a STALE chain
+	rawTx, err = tx1.EFHex()
+	require.NoError(t, err)
+	_ = postRequest[TransactionResponse](t, arcEndpointV1Tx, createPayload(t, TransactionRequest{RawTx: rawTx}), map[string]string{"X-WaitFor": StatusSeenOnNetwork}, http.StatusOK)
+
+	// generate new block that will create a stale chain that includes the txStale and tx1
+	staleHash := node_client.Generate(t, bitcoind, 1)
+
+	// verify that stale tx is still SEEN_ON_NETWORK
+	statusURL = fmt.Sprintf("%s/%s", arcEndpointV1Tx, txStale.TxID())
+	statusResp = getRequest[TransactionResponse](t, statusURL)
+	require.Equal(t, StatusSeenOnNetwork, statusResp.TxStatus)
+
+	// verify that nothing changed so far with previous mined txs
+	statusURL = fmt.Sprintf("%s/%s", arcEndpointV1Tx, tx1.TxID())
+	statusResp = getRequest[TransactionResponse](t, statusURL)
+	require.Equal(t, StatusMined, statusResp.TxStatus)
+	require.Equal(t, invHash, *statusResp.BlockHash)
+
+	statusURL = fmt.Sprintf("%s/%s", arcEndpointV1Tx, tx2.TxID())
+	statusResp = getRequest[TransactionResponse](t, statusURL)
+	require.Equal(t, StatusMined, statusResp.TxStatus)
+	require.Equal(t, tx2BlockHash, *statusResp.BlockHash)
+
+	// make the STALE chain LONGEST by adding 2 new blocks
+	node_client.Generate(t, bitcoind, 1)
+	node_client.Generate(t, bitcoind, 1)
+
+	// verify that stale tx is now MINED
+	statusURL = fmt.Sprintf("%s/%s", arcEndpointV1Tx, txStale.TxID())
+	statusResp = getRequest[TransactionResponse](t, statusURL)
+	require.Equal(t, StatusMined, statusResp.TxStatus)
+	require.Equal(t, staleHash, *statusResp.BlockHash)
+
+	// verify that previous mined tx1 have updated block info
+	statusURL = fmt.Sprintf("%s/%s", arcEndpointV1Tx, tx1.TxID())
+	statusResp = getRequest[TransactionResponse](t, statusURL)
+	require.Equal(t, StatusMined, statusResp.TxStatus)
+	require.Equal(t, staleHash, *statusResp.BlockHash)
+
+	// verify that tx2 is now MINED_IN_STALE_BLOCK
+	statusURL = fmt.Sprintf("%s/%s", arcEndpointV1Tx, tx2.TxID())
+	statusResp = getRequest[TransactionResponse](t, statusURL)
+	require.Equal(t, StatusMinedInStaleBlock, statusResp.TxStatus)
+	require.Equal(t, tx2BlockHash, *statusResp.BlockHash)
+}
```

Comment:
One additional interesting check would be to check a callback has been sent for tx2 with status `MINED_IN_STALE_BLOCK`